### PR TITLE
[WIP] APM mainnet deploy artifacts

### PR DESCRIPTION
**Pull request by izqui**

### CODE REVIEW

File: scripts/deploy-mainnet-apm.js [hunk 1]
```diff
@@ -0,0 +1,68 @@
+const namehash = require('eth-ens-namehash').hash
+const keccak256 = require('js-sha3').keccak_256
+
+const APMRegistryFactory = artifacts.require('APMRegistryFactory')
+const DAOFactory = artifacts.require('DAOFactory')
+const ENS = artifacts.require('ENS')
+
+const owner = '0x2dc1ccf1c83d6760f93adbd66997b56a267ce01a'
+const ens = '0x314159265dD8dbb310642f98f50C066173C1259b'
```
Comment (sohkai) [inline scripts/deploy-mainnet-apm.js]: Small thing: we could use [ethereum-ens-network-map](https://github.com/danfinlay/ethereum-ens-network-map) for this so we're not just throwing magic strings around.

File: scripts/deploy-mainnet-apm.js [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+  // const daoFactory = await DAOFactory.new(...daoBases, evmScriptRegistry)
+  const daoFactory = DAOFactory.at('0x7b6bddc7fa88f1340fad3b4cb0b267e0314ab76b')
+  console.log('deployed DAOFactory', daoFactory.address)
+
+  console.log('deploying APMRegistryFactory')
+  const apmFactory = await APMRegistryFactory.new(daoFactory.address, ...apmBases, ens, '0x00')
+  const apmFactory = APMRegistryFactory.at('0x66926276f7ba5d9d17015c822e0e8ad3773f20f4')
+  console.log('deployed APMRegistryFactory', apmFactory.address)
+
+  return
```
Comment (sohkai) [inline scripts/deploy-mainnet-apm.js]: ???

Should we be returning here?

File: scripts/deploy-mainnet-apm.js [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+  console.log('Deploying APM')
+  console.log('Owner:', owner)
+  console.log('ENS:', ens)
+  console.log('TLD:', tld)
+  console.log('Label:', label)
+
+  console.log('=========')
+  console.log('deploying APM bases')
+  // const apmBases = await deployBases(['APMRegistry', 'Repo', 'ENSSubdomainRegistrar'])
+  const apmBases = [ '0xf5174528211211a667243edc8579f2042124895c',
```
Comment (sohkai) [inline scripts/deploy-mainnet-apm.js]: We're generally using

```js
const apmBases = [
  '...',
  '...',
]
```

for multi-line arrays.

File: truffle-config.js [hunk 4]
```diff
@@ -12,6 +12,7 @@ if (process.env.LIVE_NETWORKS) {
   try {
     const { rpc, key } = require(require('homedir')()+'/.rinkebykey.json')
     rinkebyProvider = new PrivateKeyProvider(key, rpc)
+    mainnetProvider = new PrivateKeyProvider(key, 'https://mainnet.infura.io')
```
Comment (sohkai) [inline truffle-config.js]: This is using the same key as rinkeby?

File: truffle-config.js [hunk 5]
```diff
@@ -41,6 +42,12 @@ module.exports = {
       port: 8535,
       gas: 6.9e6,
     },
+    mainnet: {
+      network_id: 1,
+      provider: mainnetProvider,
+      gas: 7.3e6, // Lower than network limit and higher than required gas
+      gasPrice: 19000000000, // 16 gwei -> May 21st 2018, 2 * safe low
```
Comment (sohkai) [inline truffle-config.js]: **19** gwei

Comment (izqui) [reply truffle-config.js]: This was the most expensive deployment of my life. 16 was too little lol

**Comment by izqui** [comment]
Thanks for the comments @sohkai. I created the PR mostly for it to not be lost in my local repo forever, want to properly clean up so this artifacts can be easily reused and save the gas!

**Comment by sohkai** [comment]
Closing this as we've since created [deployments](https://github.com/aragon/deployments) to hold deployment information as well as deployment scripts for aragonPM in this repo.

**Pull request closed without merging**